platform/gce.go

```diff
@@ -44,10 +46,11 @@ type GCEOptions struct {
 }
 
 type gceCluster struct {
-	api      *compute.Service
-	sshAgent *network.SSHAgent
-	conf     *GCEOptions
-	machines map[string]*gceMachine
+	api        *compute.Service
+	sshAgent   *network.SSHAgent
+	conf       *GCEOptions
+	machines   map[string]*gceMachine
+	sync.Mutex // protects only against NewMachine being called in parallel
```

Comment:
Yeah, I left Destroy() alone for the same reason as Machines(). I only wanted the minimum allow NewMachine() to be called in parallel since that is the only function we currently expect to work in parallel. I'm not sure why would expect any other functions on a cluster to work in parallel since the owner of that cluster can manage that. Listing out Machines() doesn't have an expectation that they will change out from under you.

As far as associating locks with the data, are you suggesting something like this:
`type machines struct {
machs map[string]*gceMachine
sync.Mutex
}
`